### PR TITLE
Feature/integration test

### PR DESCRIPTION
**Pull request by franciscoPonceDev**
### Project requirements
-  Use Capybara to write integration tests for each view in your project.

- Login page:
  - I can see the username and password inputs and the "Submit" button.
  - When I click the submit button without filling in the username and the password, I get a detailed error.
  - When I click the submit button after filling in the username and the password with incorrect data, I get a detailed error.
  - When I click the submit button after filling in the username and the password with correct data, I am redirected to the root page.
  
- User index page:
  - I can see the username of all other users.
  - I can see the profile picture for each user.
  - I can see the number of posts each user has written.
  - When I click on a user, I am redirected to that user's show page.
  
- user show page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see the user's bio.
  - I can see the user's first 3 posts.
  - I can see a button that lets me view all of a user's posts.
  - When I click a user's post, it redirects me to that post's show page.
  - When I click to see all posts, it redirects me to the user's post's index page.

- User post index page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see a post's title.
  - I can see some of the post's body.
  - I can see the first comments on a post.
  - I can see how many comments a post has.
  - I can see how many likes a post has.
  - I can see a section for pagination if there are more posts than fit on the view.
  - When I click on a post, it redirects me to that post's show page.

- Post show page:
  - I can see the post's title.
  - I can see who wrote the post.
  - I can see how many comments it has.
  - I can see how many likes it has.
  - I can see the post body.
  - I can see the username of each commentor.
  - I can see the comment each commentor left.

### CODE REVIEW

File: spec/features/login_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    fill_in 'Password', with: '12345678'
+    click_button 'Log in'
+    expect(page).to have_content('Invalid Email or password')
+  end
+
+  it 'should redirect to root page when enters correct data' do
+    fill_in 'Email', with: 'lilly@gmail.com'
+    fill_in 'Password', with: 'lilly12345'
+    click_button 'Log in'
+    expect(page.current_path).to eql(root_path)
```
Comment (andersonlebon) [inline spec/features/login_spec.rb:30]: - [ ] Kindly note that most of your tests are not passing. Kindly fix them

image

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @franciscoPonceDev 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/login_spec.rb (hunk 1 above)
Comment (franciscoPonceDev) [reply spec/features/login_spec.rb:30]: Hello @andersonlebon , could you check if you are using your postgres credentials before running the tests? They are working fine for all members of the team.
You can change the config in your database.yml inside config folder.
image

Also, could you run:
`rails db:migrate:reset RAILS_ENV=test`
`rails db:seed RAILS_ENV=test`
before running:
`rspec spec/features`
please?

Comment (andersonlebon) [reply spec/features/login_spec.rb:30]: Let me have a look if I can rend the seed in your test DB.

**Review by BenMukebo — APPROVED** [review]
<h1>Status: Approve  :champagne:  :+1: :confetti_ball: 

### **Hi Team,** 

### **Congratulations! 🎉 Project Approved ✅**

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 


### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
testpasse


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers :clinking_glasses:  and Happy coding!!! :dancers:💻  

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/login_spec.rb (hunk 1 above)
Comment (andersonlebon) [reply spec/features/login_spec.rb:30]: ## I think your project is okay now

Comment (andersonlebon) [reply spec/features/login_spec.rb:30]: Just make sure you write instructions for running your tests in your readme

**Pull request merged**